### ISSUE TITLE
Allow augmenting the global scope with new names

### ISSUE THREAD

**Issue by yortus** [Microsoft/TypeScript#7161]
It would be great if TypeScript could model situations where new names are added to the global scope. 

This was described by a team member as a desirable feature that could definitely be enabled in https://github.com/Microsoft/TypeScript/issues/7015#issuecomment-184943914. However that issue is closed, and I can't find any other open issue tracking the suggestion, so I opened this one.

A basic example:

```typescript
// File: foo.ts

// Augment the global scope to add foo
declare global {
    var foo: Function;  // ERROR: Cannot introduce new top level names
}

// Implement foo
var fooImpl = () => {/***/};
foo = fooImpl;  // OK - compiler 'sees' global foo here
export default fooImpl;

// Reference foo
foo();          // OK - compiler 'sees' global foo here
```

Currently this produces a compiler error as shown above. Interestingly, the global `foo` *is* recognised at the two other sites where it is referenced in the example. So it appears this would already work just if the error message was suppressed.

**Comment by yortus** [comment]
Fixed by #8104?

**Comment by basarat** [comment]
> Fixed by #8104?

@yortus indeed. Just tested and it works out of the box :rose: 

![image](https://cloud.githubusercontent.com/assets/874898/14579456/2907f25e-03ed-11e6-805a-e3b73b806b25.png)

**Comment by yortus** [comment]
Sweet! Thanks for checking @basarat
